src/proofPointRegistry.ts

```diff
@@ -317,52 +340,76 @@ class ProofPointRegistry {
     proofPointId: ProofPointId
   ): Promise<ProofPointEvent[]> {
     // Prepare and store proxy object for the logic contract
-    const registry = new this._web3.eth.Contract(
-      ProofPointRegistryAbi[version].abi as any,
+    const registry = new ethers.Contract(
       logicContractAddress,
-      { data: ProofPointRegistryAbi[version].bytecode }
+      ProofPointRegistryAbi[version].abi,
+      this._provider
     );
 
-    const eventsRaw = await registry.getPastEvents("allEvents", {
-      // TODO filter doesn't work for some reason
-      // filter: {_claim: Web3.utils.keccak256(proofPointId) },
-      fromBlock: 0,
-      toBlock: "latest",
-    });
+    const filterIssued = registry.filters.Issued(
+      null,
+      this.proofPointIdToBytes(proofPointId)
+    );
+    const filterCommitted = registry.filters.Committed(
+      null,
+      this.proofPointIdToBytes(proofPointId)
+    );
+    const filterRevoked = registry.filters.Revoked(
+      null,
+      this.proofPointIdToBytes(proofPointId)
+    );
 
-    const events = eventsRaw
-      // TODO remove this client side filter once filter bug is fixed
-      .filter(
-        (ev) =>
-          ev.returnValues._claim ===
-          Web3.utils.keccak256(proofPointId.toString())
-      )
-      .filter((ev) => ev.event !== "Published")
-      .map((ev) => {
-        return {
-          blockNumber: ev.blockNumber,
-          type: this._eventNameToEventType(ev.event),
-          issuer: EthereumAddress.parse(ev.returnValues._issuer),
-          proofPointId: proofPointId,
-          transactionHash: EthereumTransactionHash.parse(ev.transactionHash),
-        };
-      });
+    const allEvents: ProofPointEvent[] = [];
+
+    allEvents.push(
+      ...(await this.getEventsByFilter(filterIssued, proofPointId))
+    );
+    allEvents.push(
+      ...(await this.getEventsByFilter(filterCommitted, proofPointId))
+    );
+    allEvents.push(
+      ...(await this.getEventsByFilter(filterRevoked, proofPointId))
+    );
 
     if (version === 1) {
-      return events;
+      return allEvents;
     }
 
-    const priorAddress = await registry.methods.getPrevious().call();
+    const priorAddress = await registry.getPrevious();
 
     const priorEvents = await this._getHistory(
       version - 1,
       priorAddress,
       proofPointId
     );
 
-    events.push(...priorEvents);
+    allEvents.push(...priorEvents);
+
+    return allEvents;
+  }
 
-    return events;
+  private async getEventsByFilter(
+    filter: ethers.EventFilter,
+    proofPointId: ProofPointId
+  ): Promise<ProofPointEvent[]> {
+    // TODO
+    // There seems to be a bug in ethers that you cannot specify the from and to block
```

Comment:
Could be handy to see if any issue raised on this....

Reply:
Couldnt find one so raised one ethers-io/ethers.js#1148

Reply:
Fixed